File: sequenceSum.js
const sequenceSum = (begin, end, step) => {
  if (begin > end) {
    return 0;
  }

  let sum = 0;
  for (let n = begin; n <= end; n += step) {
    sum += n;
  }
  return sum;
};

console.log(sequenceSum(2, 6, 2));
console.log(sequenceSum(1, 5, 1));
console.log(sequenceSum(1, 5, 3));

File: sequence_sum.rb
def sequence_sum(begin_number, end_number, step)
  return 0 if begin_number > end_number

  begin_number + sequence_sum(begin_number + step, end_number, step)
end

puts sequence_sum(1, 5, 1)
puts sequence_sum(1, 5, 3)
puts sequence_sum(0, 15, 3)
puts sequence_sum(16, 15, 3)
puts sequence_sum(2, 24, 22)
puts sequence_sum(2, 2, 2)
puts sequence_sum(2, 2, 1)
puts sequence_sum(1, 15, 3)
puts sequence_sum(15, 1, 3)